crates/ibc/src/applications/transfer/acknowledgement.rs

```diff
@@ -39,12 +39,16 @@ impl Acknowledgement {
     pub fn from_error(err: TokenTransferError) -> Self {
         Self::Error(format!("{ACK_ERR_STR}: {err}"))
     }
+
+    pub fn is_successful(&self) -> bool {
+        matches!(self, Acknowledgement::Success(_))
+    }
 }
 
 impl AsRef<[u8]> for Acknowledgement {
     fn as_ref(&self) -> &[u8] {
         match self {
-            Acknowledgement::Success(_) => ACK_SUCCESS_B64.as_bytes(),
+            Acknowledgement::Success(_) => r#"{"result":"AQ=="}"#.as_bytes(),
```

Comment:
Note that I'm still investigating whether this is correct

Reply:
If it is indeed correct, can we use the serde_json implementation instead of the hardcoded version? This would help ensure the two don't go out of sync and is a bit cleaner imho than hardcoding some JSON in there, at a small cost of efficiency though.

Reply:
What I still need to investigate is whether spaces matter in the resulting json. In the case where they do, I would prefer the hardcoded version since I don't know if `serde_json::to_string()` adds spaces or not, and I don't want to have to go check every time I forget (or be worried every time we update `serde-json`). WDYT?

Reply:
Note that [I confirmed](https://github.com/cosmos/ibc-rs/issues/369#issuecomment-1404117981) that this is the correct output.

Reply:
> What I still need to investigate is whether spaces matter in the resulting json. In the case where they do, I would prefer the hardcoded version since I don't know if serde_json::to_string() adds spaces or not, and I don't want to have to go check every time I forget (or be worried every time we update serde-json). WDYT?

Spaces don't matter; see [issue comment](https://github.com/cosmos/ibc-rs/issues/369#issuecomment-1404323569). However, what I don't like about `serde_json::to_string()` is that it returns a `Result` which we would have to `unwrap()`. Ideally, I would prefer if we didn't have any `unwrap()`s in the code, as they make audits of the code more difficult (that's an extra thing to check), and every time I update `serde-json` I'd ideally need to verify that the `Error` conditions didn't change. What's your take on this?

Reply:
Will merge for now; we can always open an issue to improve this